google-home-community.groovy

```diff
@@ -1489,38 +1491,68 @@ private deviceTraitPreferences_SensorState(deviceTrait) {
         deviceTrait.sensorTypes.each { sensorType ->
             // only display if the sensor has descriptive values
             if (GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState) {
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.availableStates",
-                    title: "Google Home Available States for ${sensorType.key}",
-                    type: "enum",
-                    multiple: "true",
-                    required: "true",
-                    options: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveState,
-                )
-                input(
-                    name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.descriptiveAttribute",
-                    title: "Hubitat Descriptive State Attribute for ${sensorType.key}",
-                    type: "text",
-                    required: "true",
-                    defaultValue: GOOGLE_SENSOR_STATES[sensorType.key].descriptiveAttribute
-                )
+                // if the sensor does not have a numeric state, do not allow the user to disable the descriptive state
+                if (GOOGLE_SENSOR_STATES[sensorType.key].numericUnits != "") {
+                    input(
+                        name: "${deviceTrait.name}.sensorTypes.${sensorType.key}.reportsDescriptiveState",
+                        title: "${sensorType.key} reports descriptive state",
+                        type: "bool",
+                        defaultValue: true,
+                        required: true,
+                        submitOnChange: true
+                    )
+                } else {
+                    deviceTrait.sensorTypes[sensorType.key].reportsDescriptiveState = true;
+                }
+                if (deviceTrait.sensorTypes[sensorType.key].reportsDescriptiveState == true) {
```

Comment:
You don't need to explicitly compare to `true`:
```suggestion
                if (deviceTrait.sensorTypes[sensorType.key].reportsDescriptiveState) {
```

Reply:
Agreed!  Fixed.